tests/integ/modin/frame/test_apply_axis_0.py

```diff
@@ -0,0 +1,608 @@
+        query_count=11,
+        high_count_expected=True,
+        high_count_reason="SNOW-1650644 & SNOW-1345395: Avoid extra caching and repeatedly creating same temp function",
+    ):
+        eval_snowpark_pandas_result(
+            snow_df, native_df, lambda x: x.apply(f, axis=0, args=(1,))
+        )
+
+    with SqlCounter(
+        query_count=11,
+        high_count_expected=True,
+        high_count_reason="SNOW-1650644 & SNOW-1345395: Avoid extra caching and repeatedly creating same temp function",
+    ):
+        eval_snowpark_pandas_result(
+            snow_df, native_df, lambda x: x.apply(f, axis=0, args=(1,), z=2)
+        )
+
+    with SqlCounter(query_count=3):
+        eval_snowpark_pandas_result(
+            snow_df,
+            native_df,
+            lambda x: x.apply(f, axis=0, args=(1,), z=2, v=3),
+            expect_exception=True,
+            expect_exception_type=SnowparkSQLException,
+            expect_exception_match="got an unexpected keyword argument",
+            assert_exception_equal=False,
+        )
+
+
+@pytest.mark.parametrize("data", [{"a": [1], "b": [2]}, {"a": [2], "b": [3]}])
+def test_apply_axis_0_with_if_where_duplicates_not_executed(data):
+    df = native_pd.DataFrame(data)
+    snow_df = pd.DataFrame(df)
+
+    def foo(x):
+        return native_pd.Series(
+            [1, 2, 3], index=["C", "A", "E"] if x.sum() > 3 else ["A", "E", "E"]
+        )
+
+    with SqlCounter(
+        query_count=11,
+        join_count=3,
+        udtf_count=2,
+        high_count_expected=True,
+        high_count_reason="SNOW-1650644 & SNOW-1345395: Avoid extra caching and repeatedly creating same temp function",
+    ):
+        eval_snowpark_pandas_result(snow_df, df, lambda x: x.apply(foo, axis=0))
+
+
+@pytest.mark.parametrize(
+    "return_value",
+    [
+        native_pd.Series(["a", np.int64(3)]),
+        ["a", np.int64(3)],
+        np.int64(3),
+    ],
+)
+@sql_count_checker(query_count=6, join_count=1, udtf_count=1)
+def test_numpy_integers_in_return_values_snow_1227264(return_value):
+    eval_snowpark_pandas_result(
+        *create_test_dfs(["a"]), lambda df: df.apply(lambda row: return_value, axis=0)
+    )
+
+
+@pytest.mark.xfail(
+    strict=True,
+    raises=SnowparkSQLException,
+    reason="SNOW-1650918: Apply on dataframe data columns containing NULL fails with invalid arguments to udtf function",
+)
+@pytest.mark.parametrize(
+    "data, apply_func",
+    [
+        [
+            [[None, "abcd"]],
+            lambda x: x + " are first 4 letters of alphabet" if x is not None else None,
+        ],
+        [
+            [[123, None]],
+            lambda x: x + 100 if x is not None else None,
+        ],
+    ],
+)
+def test_apply_axis_0_bug_1650918(data, apply_func):
+    native_df = native_pd.DataFrame(data)
+    snow_df = pd.DataFrame(native_df)
+    eval_snowpark_pandas_result(
+        snow_df,
+        native_df,
+        lambda x: x.apply(apply_func, axis=0),
+    )
```

Comment:
Is this is raising SnowparkSQLException should be catch the exception with `pytest` instead of using `eval_snowpark_pandas_result`?

Reply:
This looks like an existing bug in the underlying udtf support not handling NULL, i created the jira.  Don't we want to use xfail since it is a known failing test, rather than having it "pass".  The idea here is we don't want to refactor/rewrite the test when the issue is fixed, we simply remove the xfail annotation, ie. test driven development.

Reply:
Ah okay, I misunderstood. Yeah xfail with `eval_snowpark_pandas_result` is best here.